lib/topological_inventory/ingress_api/messaging_client.rb

```diff
@@ -0,0 +1,34 @@
+module TopologicalInventory
+  module IngressApi
+    def self.with_messaging_client
+      messaging_client ||= raw_messaging_client
+      raise if messaging_client.nil?
+
+      begin
+        yield messaging_client
+      rescue Kafka::ConnectionError
```

Comment:
looks like this one is also relevant? ```"Kafka::DeliveryFailed (Could not connect to any of the seed brokers:\n- kafka://platform-mq-ci-kafka-bootstrap.platform-mq-ci.svc:9092: Connection timed out):"``` ?

Reply:
Maybe just rescue `Kafka::Error` ?

Reply:
actually there's quite a few of the errors https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka.rb#L22

I wonder which one of them shouldn't generate new connection? Maybe lets use ```Kafka::Error``` and just filter down few we know are payload related? Like the `message too large`.

Reply:
Yeah that's what I was thinking, let me do that

Reply:
hehe ok :-D  :+1:  we always write the same comments :-D

Reply:
:smile: